features/ALB002_Alignment-layout.feature

```diff
@@ -0,0 +1,35 @@
+@implementer-agreement
+@ALB
+Feature: ALB002 - Alignment Layout
+
+  Scenario: Agreement on nested attributes of IfcAlignment
+
+      Given A file with Schema Identifier "IFC4X3"
+
+      Then Each IfcAlignment must nest exactly 1 instance(s) of IfcAlignmentHorizontal
+       And Each IfcAlignment must nest at most 1 instance(s) of IfcAlignmentVertical
+       And Each IfcAlignment must nest at most 1 instance(s) of IfcAlignmentCant  
+      # And Every oriented edge shall be referenced exactly 1 times by the loops of the face
+    
+  Scenario: Agreement on attributes being nested within a decomposition relationship
+
+      Given a file with Schema Identifier "IFC4x3"
+
+      Then Each IfcAlignmentHorizontal must be nested only by 1 IfcAlignment
+       And  Each IfcAlignmentVertical must be nested only by 1 IfcAlignment
+       And  Each IfcAlignmentCant must be nested only by 1 IfcAlignment
+    
+  Scenario: Agreement on constraints of allowed attributes nesting nested attributes of IfcAlignment
+    # @note can probably be formulated better
+
+      Given a file with Schema Identifier "IFC4x3"
+
+      Then Each IfcAlignment must nest only the following entities: IfcAlignmentHorizontal, IfcAlignmentVertical, IfcAlignmentCant, IfcReferent, IfcAlignments
```

Comment:
```suggestion
      Then Each IfcAlignment may nest only the following entities: IfcAlignmentHorizontal, IfcAlignmentVertical, IfcAlignmentCant, IfcReferent, IfcAlignment
```

- I think the meaning of `must` is too enforcing, i.e. that the entities listed `must` also be present. 
- Typo at the end, there is no `IfcAlignments` in the schema.

Reply:
I agree, this appears implemented as the entities types in the model should be a subset of the entities listed here https://github.com/buildingSMART/ifc-gherkin-rules/pull/6/files#diff-cf027576a6812e3403bd3d9579d83abfaa74a41607a4f3c986398ef8492a5d3dR248 In that case I find `may` also more suitable.

Reply:
Yes, the term 'only' seems the to be the most important/restrictive factor in this rule scenario; if this term was instead 'all', the implementation would be completely different (but changing the term 'must' to 'may' or 'should' does not change the implementation). Indeed, there should be only elements in that subset (in this case IfcAlignmentHorizontal .. AfcAlignment') in the nesting entity types (of IfcAlignment).

I've changed it to 'may' in the feature file, and thank you for noting the typo.
 I'm currently testing the scenarios with some new test files (this pull request was also partly to see the sandbox url functionality) and adding some changes. I'll also comment here when that's finished (and commit/pull request those changes)

Reply:
The changes are added in https://github.com/buildingSMART/ifc-gherkin-rules/commit/a33767f28032eb37600de5d4d0bb6834773bd60e